### DISCUSSION TITLE
Newbie question re iphone web app

### DISCUSSION THREAD

**Discussion by Dan1001** [zauberzeug/nicegui#962] [Q&A] [answered]
### Question

Hi,

nicegui looks perfect for my home automation project, but I'm struggling with how to make it run as a web app on an iPhone. I thought it would be:

```py
ui.add_head_html("<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no'>")
ui.add_head_html("<meta name='apple-mobile-web-app-capable' content='yes'>")
ui.add_head_html("<meta name='apple-mobile-web-app-tatus-bar-style' content='black'>")
```

but this doesn't seem to do anything. What am I missing? 

Thank you!

**Comment by rodja** [comment]
`ui.add_head_html("")` does nothing, because the string you pass to the function contains nothing. Could you elaborate a bit more on what you want to achieve? I would expect you to have a computer like a Raspberry Pi on which you start the NiceGUI app. Do you have trouble to access the app via iPhone?

**Reply by falkoschindler** [reply]
I just edited the original post. Due to missing code formatting the meta tags haven't been displayed.

**Comment by falkoschindler** [comment]
@Dan1001 Please explain what exactly is not working. What do you expect to happen, what is happening instead?

**Reply by Dan1001** [reply]
Thank you - apologies for the wrong formatting. I'm running the nicegui app on an ubuntu PC and want to access it on iphones, but with the ability for the user to save it to the home screen and then, on launch, have it look like an app and not a website (so no menu bar at the bottom). In HTML I achieve this with the apple-mobile-web-app-capable meta tag; but I'm not sure how to do that in nicegui. I thought ui.add_head_html() would get there, but it doesn't seem to do anything.

**Reply by falkoschindler** [reply]
Ok, I see. In principle, using `ui.add_head_html()` should be the right way to do it. You can inspect the generated source code with your browser and verify that the HTML code is inserted in the head tag.

I just passed your code to Chat GPT and it spotted a typo: "apple-mobile-web-app-tatus-bar-style" should probably be "apple-mobile-web-app-status-bar-style". Apart from that I would swap single and double quotes, since HTML usually expects double quotes, while Python doesn't care.

So I recommend fixing the typo and verifying that these three tags are actually placed in the head tag. If this is not the case, we might need to look into your code and check if maybe `ui.add_head_html()` isn't called correctly.

**Reply by Dan1001** [reply]
thank you! I've swapped the quotes round, but still doesn't work - and I can confirm the metatags aren't visible in the HTML.

The full code looks like this:

```py
from fastapi import Request
from nicegui import ui


@ui.page('/')
def main_page(request: Request) -> None:
    ui.button('Test button')
    

ui.add_head_html('<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">')
ui.add_head_html('<meta name="apple-mobile-web-app-capable" content="yes">')
ui.add_head_html('<meta name="apple-mobile-web-app-status-bar-style" content="black">')

ui.run(dark=True, title="Test page")

```

Am I doing something dumb or is there a bug?

**Reply by rodja** [reply, marked as answer]
Ahhh... you need to indent the `ui.add_head_html` calls to make them part of the `main_page`. The calls are not global but only apply to the current page.

**Reply by Dan1001** [reply]
brilliant - thank you!

**Reply by Dan1001** [reply]
just to add to this for anyone searching - to make the apple-touch-icon work I added:

```py
from nicegui import ui, app
app.add_static_files('/static', 'static')

ui.add_head_html('<link rel="apple-touch-icon" href="static/apple-touch-icon.png">')
```

and put the png file in the static directory.

**Reply by falkoschindler** [reply]
Thanks, @Dan1001!
I'll leave a reference to how nicegui.io provides icons for various browsers and phones:
https://github.com/zauberzeug/nicegui/blob/287455511e079909bf9046068f8d9835d623fe87/website/static/header.html#L7-L15

**Reply by Dan1001** [reply]
thanks! Although the bit that confused me was where to put the png files. With the app.add_static line I can then put them in a "static" directory. Does your reference require something similar, or is there an easier way I'm missing?

**Reply by falkoschindler** [reply]
The "/favicon" route is also added using `add_static_files`:
https://github.com/zauberzeug/nicegui/blob/287455511e079909bf9046068f8d9835d623fe87/main.py#L30
